File: go/support_ticket.go
package zymeup

type SupportTicketClient struct {
	client *Client
}

func NewSupportTicketClient(c *Client) *SupportTicketClient {
	return &SupportTicketClient{client: c}
}

func (c *SupportTicketClient) Create(data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/support/tickets", data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *SupportTicketClient) List(params map[string]interface{}) ([]map[string]interface{}, error) {
	path := "/api/v1/support/tickets" + buildQueryString(params)
	resp, err := c.client.doRequest("GET", path, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int `json:"code"`
		Data struct {
			Items []map[string]interface{} `json:"items"`
			Total int                      `json:"total"`
		} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data.Items, nil
}

func (c *SupportTicketClient) Get(id string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("GET", "/api/v1/support/tickets/"+id, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *SupportTicketClient) AddMessage(id string, content string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/support/tickets/"+id+"/messages", map[string]interface{}{
		"content": content,
	})
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

File: node/src/support_ticket.ts
import { HttpClient } from './http-client';
import type { ZymeupConfig } from './http-client';
import type { ApiResult } from './index';

// ============ Types ============

export type TicketPriority = 'low' | 'normal' | 'high' | 'urgent';
export type TicketStatus = 'open' | 'closed' | 'pending';

export interface SupportTicketMessage {
    id: string;
    content: string;
    sender_name: string;
    sender_id: string;
    created_at: string;
}

export interface SupportTicket {
    id: string;
    subject: string;
    description: string;
    priority: TicketPriority;
    status: TicketStatus;
    tenant_id: string;
    tenant_type: string;
    owner_id: string;
    owner_name: string;
    created_at: string;
    updated_at: string;
    messages?: SupportTicketMessage[];
}

export interface CreateTicketRequest {
    subject: string;
    description: string;
    priority?: TicketPriority;
}

export interface ListTicketsFilter {
    status?: TicketStatus;
    page?: number;
    page_size?: number;
}

export interface TicketListResponse {
    data: SupportTicket[];
    total: number;
    page: number;
    page_size: number;
}

// ============ Client ============

export class SupportTicketClient extends HttpClient {
    constructor(config: Partial<ZymeupConfig> = {}) {
        super(config);
    }

    async create(req: CreateTicketRequest): Promise<ApiResult<SupportTicket>> {
        return this.request('/api/v1/shipment/support/tickets', 'POST', req);
    }

    async list(filter: ListTicketsFilter = {}): Promise<ApiResult<TicketListResponse>> {
        const params: Record<string, string> = {};
        if (filter.status) params.status = filter.status;
        if (filter.page) params.page = String(filter.page);
        if (filter.page_size) params.page_size = String(filter.page_size);
        const q = this.buildQuery(params);
        return this.request(`/api/v1/shipment/support/tickets${q}`);
    }

    async get(id: string): Promise<ApiResult<SupportTicket>> {
        return this.request(`/api/v1/shipment/support/tickets/${encodeURIComponent(id)}`);
    }

    async addMessage(id: string, content: string): Promise<ApiResult<SupportTicketMessage>> {
        return this.request(`/api/v1/shipment/support/tickets/${encodeURIComponent(id)}/messages`, 'POST', { content });
    }
}
